pkg/machine/qemu/command.go

```diff
@@ -0,0 +1,100 @@
+	q := QemuCmd{binary}
+	q = append(q, options...)
+	q = propagateHostEnv(q)
+	return &q
+}
+
+// WithMemory adds the specified amount of memory for the machine
+func (q *QemuCmd) WithMemory(m uint64) *QemuCmd {
+	*q = append(*q, "-m", strconv.Itoa(int(m)))
+	return q
+}
+
+// WithCPUs adds the number of CPUs the machine will have
+func (q *QemuCmd) WithCPUs(c uint64) *QemuCmd {
+	*q = append(*q, "-smp", strconv.Itoa(int(c)))
+	return q
+}
+
+// WithIgnitionFile specifies the machine's ignition file
+func (q *QemuCmd) WithIgnitionFile(file machine.VMFile) *QemuCmd {
+	*q = append(*q, "-fw_cfg", "name=opt/com.coreos/config,file="+file.GetPath())
+	return q
+}
+
+// WithQmpMonitor specifies the machine's qmp socket
+func (q *QemuCmd) WithQmpMonitor(monitor Monitor) *QemuCmd {
+	*q = append(*q, "-qmp", monitor.Network+":"+monitor.Address.GetPath()+",server=on,wait=off")
+	return q
+}
+
+// WithNetwork adds a network device to the machine
+func (q *QemuCmd) WithNetwork() *QemuCmd {
+	// Right now the mac address is hardcoded so that the host networking gives it a specific IP address.  This is
+	// why we can only run one vm at a time right now
+	*q = append(*q, "-netdev", "socket,id=vlan,fd=3", "-device", "virtio-net-pci,netdev=vlan,mac=5a:94:ef:e4:0c:ee")
+	return q
+}
+
+// WithSerialPort adds a serial port to the machine for readiness
+func (q *QemuCmd) WithSerialPort(readySocket, vmPidFile machine.VMFile, name string) *QemuCmd {
+	*q = append(*q,
+		"-device", "virtio-serial",
+		// qemu needs to establish the long name; other connections can use the symlink'd
+		// Note both id and chardev start with an extra "a" because qemu requires that it
+		// starts with a letter but users can also use numbers
+		"-chardev", "socket,path="+readySocket.GetPath()+",server=on,wait=off,id=a"+name+"_ready",
+		"-device", "virtserialport,chardev=a"+name+"_ready"+",name=org.fedoraproject.port.0",
+		"-pidfile", vmPidFile.GetPath())
+	return q
+}
+
+// WithVirtfsMount adds a virtfs mount to the machine
+func (q *QemuCmd) WithVirtfsMount(source, tag, securityModel string, readonly bool) *QemuCmd {
+	virtfsOptions := fmt.Sprintf("local,path=%s,mount_tag=%s,security_model=%s", source, tag, securityModel)
+	if readonly {
+		virtfsOptions += ",readonly"
+	}
+	*q = append(*q, "-virtfs", virtfsOptions)
+	return q
+}
+
+// WithBootableImage specifies the image the machine will use to boot
+func (q *QemuCmd) WithBootableImage(image string) *QemuCmd {
+	*q = append(*q, "-drive", "if=virtio,file="+image)
+	return q
+}
+
+// WithDisplay specifies whether the machine will have a display
+func (q *QemuCmd) WithDisplay(display string) *QemuCmd {
+	*q = append(*q, "-display", display)
+	return q
+}
+
+// WithArchOptions adds any architecture-specific options to the machine including
+// image location
+func (q *QemuCmd) WithArchOptions(options []string) *QemuCmd {
+	*q = append(*q, options...)
+	return q
+}
+
+func (q *QemuCmd) Build() QemuCmd {
```

Comment:
should this return []string ?

Reply:
Yes, I believe you're right. I will fix that